include/igraph_error.h

```diff
@@ -469,6 +469,36 @@ typedef enum {
 DECLDIR int igraph_error(const char *reason, const char *file, int line,
                          int igraph_errno);
 
+/**
+ * \define IGRAPH_ERRORF
+ * \brief Trigger an error, with printf-like error output.
+ *
+ * \a igraph functions can use this macro when they notice an error and
+ * want to pass on extra information to the user about what went wrong.
+ * It calls \ref igraph_errorf() with the proper parameters and if that
+ * returns the macro returns the "calling" function as well, with the
+ * error code. If for some (suspicious) reason you want to call the
+ * error handler without returning from the current function, call
+ * \ref igraph_errorf() directly.
+ * \param reason Textual description of the error, a template string
+ *   with the same syntax as the standard printf C library function.
+ *   This should be something more descriptive than the text associated
+ *   with the error code. E.g. if the error code is \c IGRAPH_EINVAL,
+ *   its associated text (see  \ref igraph_strerror()) is "Invalid
+ *   value" and this string should explain which parameter was invalid
+ *   and maybe what was expected and what was recieved.
+ * \param igraph_errno The \a igraph error code.
+ * \param ... The additional arguments to be substituted into the
+ *   template string.
+ */
+
+#define IGRAPH_ERRORF(reason,igraph_errno, ...) \
```

Comment:
Nitpick: can you add a space after `reason,`?

Reply:
Added them myself. The space was missing in two other places as well.